File: src/algorithms/sort/ts/insertionSort.ts
export function insertionSort(values: number[]): void {
    if (values.length <= 1) {
        return
    }

    for (let i = 1; i < values.length; i++) {
        let indexToInsert = i
        const valueToInsert = values[i]

        for (let j = i - 1; j >= 0; j--) {
            if (valueToInsert < values[j]) {
                values[j + 1] = values[j]
                indexToInsert = j
            } else {
                break
            }
        }

        values[indexToInsert] = valueToInsert
    }
}

File: src/algorithms/sort/c/insertionSort.c
#include <stdio.h>

int insertionSort(int *arr, int arrLength) {
    if(arrLength <= 0){
        printf("Array length must be greater than 0");
        return -1;
    } else if (arrLength == 1){
        return 0;
    }

    for (int i = 1; i < arrLength; i++) {
        int valueToInsert = arr[i];
        int j = i -1 ;

        while(j >= 0 && valueToInsert < arr[j]){
            arr[j + 1] = arr[j];
            j -= 1;
        }

        arr[j+1] = valueToInsert;
    }

    return 0;
}
